packages/rocketchat-livechat/imports/server/rest/visitors.js

```diff
@@ -30,3 +30,35 @@ RocketChat.API.v1.addRoute('livechat/visitor/:visitorToken/room', { authRequired
 		return RocketChat.API.v1.success({ rooms });
 	}
 });
+
+RocketChat.API.v1.addRoute('livechat/visitor/:visitorToken/room/close', { authRequired: true }, {
+	post() {
+		if (!RocketChat.authz.hasPermission(this.userId, 'view-livechat-manager')) {
+			return RocketChat.API.v1.unauthorized();
+		}
+
+		const visitor = LivechatVisitors.getVisitorByToken(this.urlParams.visitorToken);
+		if (!visitor) {
+			return RocketChat.API.v1.failure('Visitor not found');
+		}
+
+		const rooms = RocketChat.models.Rooms.findOpenByVisitorToken(this.urlParams.visitorToken).fetch();
+		if (!rooms || rooms.length === 0) {
+			return RocketChat.API.v1.failure('Visitor dont have any opened room');
+		}
+		const room = rooms[0];
+
+		const user =
```

Comment:
Can you give more details on which situation you need to use `username` instead of `this.userId`?
I mean, the current user is the agent that is participating the chat?